src/store-hooks.ts

```diff
@@ -0,0 +1,56 @@
+import { useEffect, useRef, useState } from 'react';
+import { Store } from './store';
+
+/**
+ * @param {Store} store  - Store to subscribe.
+ * @param {Function} callback - Function to call on each dipatch.
+ * @param {Function} errorcCallback - Function to call on each error dipatch.
+ */
+export function useStoreSub<T>(
```

Comment:
why not: 
```suggestion
export function useStore<T>(
```

Reply:
This hook is to handle callbacks subscribed to the store. It is not the best pattern to use but allows us to build other hooks that are more useful